addons/website/controllers/main.py

```diff
@@ -388,11 +388,13 @@ def save_session_layout_mode(self, layout_mode, view_id):
         request.session[f'website_{view_id}_layout_mode'] = layout_mode
 
     @http.route('/website/snippet/filters', type='json', auth='public', website=True)
-    def get_dynamic_filter(self, filter_id, template_key, limit=None, search_domain=None, with_sample=False):
-        dynamic_filter = request.env['website.snippet.filter'].sudo().search(
-            [('id', '=', filter_id)] + request.website.website_domain()
-        )
-        return dynamic_filter and dynamic_filter._render(template_key, limit, search_domain, with_sample) or []
+    def get_dynamic_filter(
+        self, filter_id, template_key, limit=None, search_domain=None, with_sample=False,
+        show_variants=True,
+    ):
+        dynamic_filter = request.env['website.snippet.filter'].sudo().search([('id', '=', filter_id)] + request.website.website_domain())
+        rendered_dynamic_filter = dynamic_filter._render(template_key, limit, search_domain, with_sample, show_variants)
```

Comment:
I took into consideration your comments, and I implemented the new changes
However, regarding using a `super` call from the overridden route in the controller, I don't think its entirely applicable here because the original `get_dynamic_filter` returns a list and I need to apply `with_context` on the model.
Also the `_render` method is a `website` method and relevant methods to `website_sale` are overridden when needed.
Let me know if you see any other way it can improve :)